packages/test/src/workflows/unblock-or-cancel.ts

```diff
@@ -1,9 +1,20 @@
+/**
+ * All-in-one sample showing cancellation, signals and queries
+ * @module
+ */
 // @@@SNIPSTART nodejs-blocked-workflow
 import { Trigger, CancelledFailure } from '@temporalio/workflow';
 import { Blocked } from '../interfaces';
 
+let blocked = true;
 const unblocked = new Trigger<void>();
 
+const queries = {
+  isBlocked(): boolean {
+    return blocked;
```

Comment:
It might be neater if triggers had an `isBlocked()` method you could use instead of having to hard-code a `let blocked` variable.

Reply:
`Trigger`s can do more than just unblock the workflow but I agree we could expose the result of the trigger.
Something along the lines of:

```ts
const blocked = new Trigger<boolean>(true);
blocked.value; // true
blocked.resolve(false);
blocked.value; // false
// -- or --
blocked.reject(new Error('err'));
blocked.value // throws Error: err
```